lib/ui/platform_dispatcher.dart

```diff
@@ -308,6 +308,116 @@ class PlatformDispatcher {
     _invoke(onMetricsChanged, _onMetricsChangedZone);
   }
 
+  // A debug-only variable that stores the [FlutterView]s for which
+  // [FlutterView.render] has already been called during the current
+  // [onBeginFrame]/[onDrawFrame] callback sequence.
+  //
+  // It is null outside the scope of those callbacks indicating that calls to
+  // [FlutterView.render] must be ignored. Furthermore, if a given [FlutterView]
+  // is already present in this set when its [FlutterView.render] is called
+  // again, that call must be ignored as a duplicate.
+  //
+  // Between [onBeginFrame] and [onDrawFrame] the properties value is
+  // temporarily stored in `_renderedViewsBetweenCallbacks` so that it survives
+  // the gap between the two callbacks.
+  //
+  // In release build, this variable is null, and therefore the calling rule is
+  // not enforced. This is because the check might hurt cold startup delay;
+  // see https://github.com/flutter/engine/pull/46919.
+  Set<FlutterView>? _debugRenderedViews;
+  // A debug-only variable that temporarily stores the `_renderedViews` value
+  // between `_beginFrame` and `_drawFrame`.
+  //
+  // In release build, this variable is null.
+  Set<FlutterView>? _debugRenderedViewsBetweenCallbacks;
```

Comment:
Are the above part of focus handling? They look a bit like merge conflicts.

Reply:
Great catch. Yeah, I did something wrong while rebasing. Fixed it!